### PR TITLE
make pull master and agent ports configurable

### PR DESCRIPTION
**Pull request by darkonie**

### CODE REVIEW

File: api/pull.go [hunk 1]
```diff
@@ -556,9 +556,18 @@ func collectResponses(respChan <-chan *httpResponse, totalHosts int) (responses
...
+		var (
+			response httpResponse
+			port int
+		)
+
+		if host.Role == MasterRole {
+			port = dt.Cfg.FlagMasterPort
+		} else {
+			port = dt.Cfg.FlagAgentPort
+		}
```
Comment (malnick) [inline api/pull.go]:
```port = dt.Cfg.FlagAgentPort
if host.Role == MasterRole {
    port = dt.Cfg.FlagMasterPort
}```

File: api/snapshot.go [hunk 2]
```diff
@@ -230,8 +230,15 @@ func (j *SnapshotJob) runBackgroundJob(nodes []Node, config *Config, DCOSTools D
 	defer j.Unlock()
 
 	for _, node := range nodes {
+		var port int
+		if node.Role == MasterRole {
+			port = config.FlagMasterPort
+		} else {
+			port = config.FlagAgentPort
+		}
+
```
Comment (malnick) [inline api/snapshot.go]: Maybe the same as above?

File: api/snapshot.go [hunk 3]
```diff
@@ -696,24 +703,36 @@ func loadExternalProviders(config *Config) (externalProviders LogProviders, err
...
+	if err != nil {
+		return internalConfigProviders, err
+	}
+
+	if role == MasterRole {
+		port = config.FlagMasterPort
+	} else {
+		port = config.FlagAgentPort
+	}
+
```
Comment (malnick) [inline api/snapshot.go]: Here too.

File: api/snapshot.go [hunk 4]
```diff
@@ -735,6 +755,12 @@ func (j *SnapshotJob) getLogsEndpoints(config *Config, DCOSTools DCOSHelper) (en
 		log.Error("Failed to get a current role for a config")
 	}
 
+	if currentRole == MasterRole {
+		port = config.FlagMasterPort
+	} else {
+		port = config.FlagAgentPort
+	}
+
```
Comment (malnick) [inline api/snapshot.go]: Here too.

**Comment by malnick** [comment]
Extremely minor comment to get rid of `else` statement. Aside from that 🚢

**Comment by darkonie** [comment]
@malnick fixed, please review againt

**Comment by malnick** [comment]
🚢

**Pull request merged**